winch/codegen/src/masm.rs

```diff
@@ -478,6 +479,17 @@ impl Imm {
             Self::V128(_) => OperandSize::S128,
         }
     }
+
+    /// Get a little endian representation of the immediate.
+    pub fn to_bytes(&self) -> Vec<u8> {
+        match self {
+            Imm::I32(n) => n.to_le_bytes().to_vec(),
+            Imm::I64(n) => n.to_le_bytes().to_vec(),
+            Imm::F32(n) => n.to_le_bytes().to_vec(),
+            Imm::F64(n) => n.to_le_bytes().to_vec(),
+            Imm::V128(n) => n.to_le_bytes().to_vec(),
```

Comment:
Could you return a `slice` here instead? The constant pool will copy the bytes into a `Vec<u8>`, but returning a slice will at least prevent heap allocations in case this gets used for anything else that's not constant pool related.

Reply:
To be more specific, using `as_slice()` instead of `to_vec()`, similar to the other uses of `add_constant`.

Reply:
If I use `as_slice()` and return a `&[u8]`, I get this message from the compiler: `cannot return value referencing temporary value`. Likely because the array that's created by `to_le_bytes()` gets dropped when the method returns. The arrays are also of different lengths between the variants. We could maybe use some unsafe code to create a reference to the underlying bytes. Otherwise I'm not sure how to avoid the heap allocation.

Reply:
Given that this is not used for anything that the constant pool, I don't think we need to introduce any unsafe code. Perhaps adding a comment stating that this method heap allocates and it's intended to be used mostly with the constant pool?